src/Shared/CompatibilityHelpers.cs

```diff
@@ -66,4 +66,26 @@ public static CancellationTokenRegistration RegisterWithCancellationTokenCallbac
         return cancellationToken.Register((state) => callback(state, cancellationToken), state);
 #endif
     }
+
+    public static Task<T> AwaitWithYieldAsync<T>(Task<T> callTask)
+    {
+        // A completed task doesn't need to yield because code after it isn't run in a continuation.
+        if (callTask.IsCompletedSuccessfully())
```

Comment:
Why not `IsCompleted`?

Reply:
Good point. The particular task we want to force this behavior on `GrpcCall.CallTask` will never throw, but this code ended up being split out into a helper method that could technically be used with any task.

I'll change it.